star/private/star_utils.f90

```diff
@@ -2813,12 +2813,12 @@ subroutine weighed_smoothing(dd, n, ns, preserve_sign, ddold)
             end if
          end do
 
-      end subroutine weighed_smoothing
+      end subroutine weighted_smoothing
 
 
-      subroutine threshold_smoothing (dd, dd_thresh, n, ns, preserve_sign, ddold)
+      subroutine threshold_smoothing(dd, dd_thresh, n, ns, preserve_sign, ddold)
 
-        ! Same as weighed_smoothing, but only smooth contiguous regions where |dd| >= dd_thresh
+        ! Same as weighted_smoothing, but only smooth contiguous regions where |dd| >= dd_thresh
```

Comment:
The code looks fine as is but as a low-priority remark, this smells like the two functions could be one, possibly by implementing `dd_thresh` as an optional argument. If absent (`.not. present(...)`), setting `dd_thresh` to +∞ presumably has the same effect (though not sure about NaNs) as not setting `dd_thresh` at all.